angular/projects/global/src/lib/utils/utils.ts

```diff
@@ -1,3 +1,11 @@
+const encode = encodeURIComponent;
+window['encodeURIComponent'] = (component: string) => {
+  return encode(component).replace(/[!'()*]/g, c => {
+    // Also encode !, ', (, ), and *
+    return '%' + c.charCodeAt(0).toString(16);
+  });
+};
+
```

Comment:
Similar question further below.

Any reason not to use something like: 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

to handle encoding?